codice/ddf
# DDF-5535 improve DynamicSchemaResolver so that is pre-populates its cache with all metacard types

#### What does this PR do?
Improve DynamicSchemaResolver so that is pre-populates its cache with all metacard types. This is done by creating a reference listener in blueprint for metacard types and adding/removing those types from the resolver.

#### Who is reviewing it? 
@pklinef 
@rzwiefel 
@derekwilhelm 

#### Select relevant component teams: 
@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@bdthomson

#### How should this be tested?
The original problem can be reproduced by first installing a master build. Then, using the karaf console, run `catalog:search --cql '"security.access-groups" IS NULL'`. You should see an exception in the logs with the message `Anonymous Field Property does not exist. security.access-groups`.

With this PR, you should no longer see the exception.

#### Any background context you want to provide?

This is affecting downstream projects.

#### What are the relevant tickets?
Fixes: #5535 

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
